### PR TITLE
Fix ebusd service argument passing

### PR DESCRIPTION
**Pull request by bobrippling**
## Description of changes

This resolves a bug introduced in #308500 / #264893, where ebusd arguments were changed from `--scanconfig=all` to `--scanconfig all` with the use of `cli.toGNUCommandLineShell`.

The bug is an unfortunate result of `ebusd`'s argument parsing - the arguments must be [separated by equals signs](https://github.com/john30/ebusd/blob/c084dd2cbb3411883bd1b048585724a93e1aff83/src/ebusd/main_args.cpp#L658).

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bobrippling** [comment]
I'm not familiar with the nix language and suspect this fix may not be the best approach, open to suggestions.

@nathan-gs are you able to recommend anyone that might help with improving my fix?

**Comment by nathan-gs** [comment]
For me it looks good! However let's also have a review from some experts of `cli.toGNUCommandLineShell`, just to be sure (@Gabriella439 / @Profpatsch)?

**Review by nathan-gs — APPROVED** [review]

**Review by Gabriella439 — APPROVED** [review]

File: lib/cli.nix [hunk 1]
```diff
@@ -90,15 +90,33 @@ rec {
     mkOption ?
       k: v: if v == null
             then []
-            else [ (mkOptionName k) (lib.generators.mkValueStringDefault {} v) ]
+            else [ (mkOptionName k) (lib.generators.mkValueStringDefault {} v) ],
```
Comment (infinisil) [inline lib/cli.nix:93]: The current implementation looks rather backwards. Shouldn't something simple like

```suggestion
            else if optionValueSeparator == null then
              [ (mkOptionName k) (lib.generators.mkValueStringDefault {} v) ]
            else
              [ "${mkOptionName k}${optionValueSeparator}${lib.generators.mkValueStringDefault {} v}" ],
```

work?

**Review by infinisil — CHANGES_REQUESTED** [review]

File: lib/cli.nix (hunk 1 above)
Comment (bobrippling) [reply lib/cli.nix]: Thank you, yes that's much better

**Review by nathan-gs — APPROVED** [review]

File: lib/cli.nix [hunk 2]
```diff
@@ -90,7 +90,16 @@ rec {
...
-            else [ (mkOptionName k) (lib.generators.mkValueStringDefault {} v) ]
+            else if optionValueSeparator == null then
+              [ (mkOptionName k) (lib.generators.mkValueStringDefault {} v) ]
+            else
+              [ "${mkOptionName k}${optionValueSeparator}${lib.generators.mkValueStringDefault {} v}" ],
+
+    # how to separate an option from its flag;
+    # by default, there is no separator, so option `-c` and value `5`
+    # would become ["-c" "5"].
+    # This is useful if the command requires equals, for example, `-c=5`.
```
Comment (infinisil) [inline lib/cli.nix:101]: Unfortunately this is currently not rendered in the reference manual. If you're interested in improving that, feel free to make a PR to move these into the doc comment above the function! Check out the part on 'function arguments' [here](https://github.com/NixOS/nixpkgs/tree/master/doc#documentation-conventions) to see the convention for that :)

**Review by infinisil — APPROVED** [review]
Looking good!

File: lib/cli.nix (hunk 2 above)
Comment (bobrippling) [reply lib/cli.nix:101]: Thanks for the tip - would you mind taking a look at #315820? Also, do you have a link to the reference doc? I can't find where `cli.nix` appears when I've been searching

**Pull request merged**